### PR TITLE
[stable/dex] Configured more specific roles and cluster roles

### PR DESCRIPTION
**Pull request by plnordquist-pnnl**
#### What this PR does / why we need it:

Change the ClusterRoleBinding to be more specific than cluster-admin, based on the ClusterRole at https://github.com/dexidp/dex/blob/master/examples/k8s/dex.yaml. Also add a Role to allow the cert jobs to create and delete their configmaps and secrets.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/dex/templates/clusterrole.yaml
Comment (lachie83) [inline stable/dex/templates/clusterrole.yaml]: I suggest we either go to the stable RBAC API version or introduce a helper like so. I believe all support k8s versions are now on v1 stable RBAC API
```
{{/*
Return the appropriate apiVersion for RBAC APIs.
*/}}
{{- define "rbac.apiVersion" -}}
{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1" -}}
rbac.authorization.k8s.io/v1
{{- else if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1beta1" -}}
rbac.authorization.k8s.io/v1beta1
{{- else -}}
rbac.authorization.k8s.io/v1alpha1
{{- end -}}
{{- end -}}
```

Comment (rendhalver) [reply stable/dex/templates/clusterrole.yaml]: If that gives us compatibility with all supported versions then I think that's a good plan.

Comment (plnordquist-pnnl) [reply stable/dex/templates/clusterrole.yaml]: It looks like RBAC API v1 was introduced in k8s 1.8 and from what I can tell k8s supports 3 released versions so for now that would be v1.13, v1.12 and v1.11.  With that in mind, I'll update this to RBAC API v1 and skip the helper.  If anyone had a deployed version of the chart, they would need to update k8s to a supported version before updating the chart.

**Comment by plnordquist** [comment]
While I was reviewing RBAC API versions, I found that the Deployment is using a beta version as well.  I'm not sure changing that belongs in this PR but it seems like a small change to put in its own PR. The apps/v1 API version was released in k8s 1.9.

**Review by sstarcher — APPROVED** [review]

File: stable/dex/Chart.yaml
Comment (cpanato) [inline stable/dex/Chart.yaml]: this is not a bug fix, it is adding new functionality. I would say bump to 0.7.0. However, since this chart is in the stable folder I vote to turn that 1.0.0, as we had discussed some time ago in a charts meetings. @lachie83 any thoughts?

Comment (plnordquist-pnnl) [reply stable/dex/Chart.yaml]: I've updated the chart version to 0.7.0. I'll wait for further discussion on if the chart version needs to be updated again.

**Comment by plnordquist** [comment]
I've updated the deployment API version to apps/v1. I missed the signoffs on my new commits and naively ran the rebase command that the docs suggested so it changed all 4 commits. Next time I'll make sure to fix only the commits that didn't have the signoff.

**Comment by plnordquist** [comment]
I've rebased the changes so they apply cleanly to master and renamed one of the commits to remove the specific version from it.

**Comment by rendhalver** [comment]
/hold
What is the reasoning behind upgrading the chart to v1 in this PR?
Is it necessary for the changes you are making?
This will break peoples deploys if they are using a version of Kube that doesn't support apps/v1.
I think we should move that change into it's own PR rather than pollute this PR.

**Comment by sstarcher** [comment]
@rendhalver what's the policy for helm with updating API versions?  Kubernetes only supports 3 versions back which means any cluster that does not support apps v1 would be EOL already.

**Comment by plnordquist** [comment]
I can put the update to apps/v1 in a different PR. There was talk of making the chart v1.0 so I was trying to clean it up in this PR. I didn't hear anything back on updating the version to 1.0 so I didn't change that part.

**Comment by paulczar** [comment]
looks like the change to API versions has been backed off here so I'm going to cancel the hold and lgtm.

/hold cancel
/lgtm

**Pull request merged**
